sklearn/datasets/_arff_parser.py

```diff
@@ -376,6 +377,30 @@ def _pandas_arff_parser(
     columns_to_keep = [col for col in frame.columns if col in columns_to_select]
     frame = frame[columns_to_keep]
 
+    # `pd.read_csv` only consider double quotes as delimiters for strings.
+    # In case that a single quote is used as a delimiter, we need to strip it.
+    pattern = r"^(?P<quotationsymbol>['])(?P<contents>.*)(?P=quotationsymbol)$"
```

Comment:
I guess you can simplify the regexp now that we are only looking for single quotes:
```suggestion
    pattern = r"^'(?P<contents>.*)'$"
```